src/variable-source.js

```diff
@@ -0,0 +1,235 @@
+/**
```

Comment:
Hmm. Maybe you are right. I was mostly reacting that it's only used by the service or service-depending code. But we can keep it here for now.